core/shared/platform/common/posix/posix_memmap.c

```diff
@@ -281,6 +285,10 @@ os_mprotect(void *addr, size_t size, int prot)
     if (prot & MMAP_PROT_EXEC)
         map_prot |= PROT_EXEC;
 
+    if (mprotect(addr, request_size, map_prot) == -1) {
```

Comment:
L292 also calls mprotect again, seems we should change it to `return 0`?
And here is it better to change `== -` to `! = 0`?

Reply:
My bad, `os_mprotect` shouldn't be modified, I forgot to delete those lines